src/edge_detection_helper.h

```diff
@@ -0,0 +1,18 @@
+#pragma once

```

Comment:
Please put this code into `edge_detection.h` file because it's a bad practice to put declaration and implementation in 2 differently named files and this is not something an user would understand why we he/she needs to use helper file as well.